ksql-parser/src/main/java/io/confluent/ksql/parser/SchemaParser.java

```diff
@@ -83,16 +86,31 @@ public void syntaxError(
 
     final SqlTypeParser typeParser = SqlTypeParser.create(typeRegistry);
 
-    final List<TableElement> elements = parser.tableElements().tableElement()
+    final List<TableElement> elements = parser.schemaTableElements().schemaTableElement()
```

Comment:
It might be worth splitting `SchemaParser` into more strongly typed `LogicalSchemaParser` and `TableElementsParser`